pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -653,7 +695,12 @@ allChainsLabel:
 		allChains = append(allChains, allChains...)
 		if tlsInspectorEnabled {
 			allChains = append(allChains, plugin.FilterChain{})
-			filterChainMatchOption = inboundPermissiveFilterChainMatchOptions
+			if util.IsTCPMetadataExchangeEnabled(node) {
```

Comment:
Could this be a problem if the clients of this server has not upgraded?

Reply:
It should not be because, then ALPN won't match and metadata exchange won't get enabled.